tests/checks/test_check_CountMarkdownTags.py

```diff
@@ -0,0 +1,490 @@
+        assert check_exists is True
+        check = checker_source.load_plugin(check_file)
+        check_result = check.act(parsed_arguments, remaining_arguments)
+        # check the result
+        assert check_result is not None
+        assert len(check_result) == 1
+        # assert check_result[0] is expected_result
+        # check the contents of the report
+        assert report.get_result() is not None
+        assert len(report.get_result()["check"]) > 1
+        assert report.get_result()["outcome"] is expected_result
+        if expected_result:
+            assert report.get_result()["diagnostic"] == ""
+        else:
+            assert report.get_result()["diagnostic"] != ""
+
+
+@pytest.mark.parametrize(
+    "commandline_arguments, chosen_file, containing_directory, provided_count, expected_result",
+    [
+        (
+            [
+                "CountMarkdownTags",
+                "--file",
+                "file_to_find",
+                "--directory",
+                "containing_directory",
+                "--count",
+                "5",
+                "--tag",
+                "heading",
+            ],
+            "file_to_find",
+            "containing_directory",
+            "5",
+            True,
+        ),
+        (
+            [
+                "CountMarkdownTags",
+                "--file",
+                "file_to_find",
+                "--directory",
+                "containing_directory",
+                "--count",
+                "100",
+                "--tag",
+                "heading",
+            ],
+            "file_to_find",
+            "containing_directory",
+            "100",
+            False,
+        ),
+        (
+            [
+                "CountMarkdownTags",
+                "--file",
+                "file_to_find",
+                "--directory",
+                "containing_directory",
+                "--count",
+                "5",
+                "--exact",
+                "--tag",
+                "heading",
+            ],
+            "file_to_find",
+            "containing_directory",
+            "5",
+            True,
+        ),
+        (
+            [
+                "CountMarkdownTags",
+                "--file",
+                "file_to_find",
+                "--directory",
+                "containing_directory",
+                "--count",
+                "100",
+                "--exact",
+                "--tag",
+                "heading",
+            ],
+            "file_to_find",
+            "containing_directory",
+            "100",
+            False,
+        ),
+    ],
+)
+def test_act_produces_output_with_exact(
+    commandline_arguments,
+    chosen_file,
+    containing_directory,
+    provided_count,
+    expected_result,
+    tmpdir,
+):
+    """Check that using the check produces output."""
+    test_contents = """
+## What Do People Think about GatorGrader?
```

Comment:
This should probably be a module-level constant instead of specified in individual tests, unless there are substantial differences that can't be added on in specific tests.